### ISSUE TITLE
File collisions between branded and non-branded clients since ownCloud 3

### ISSUE THREAD

**Issue by olifre** [owncloud/client#10965]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the bug

Installing a branded client starting with version 3 yields the following installed files which do not contain the branded name:
```
 *      /usr/include/ownCloud/common/utility.h                                                                                                                                                                                              
 *      /usr/include/ownCloud/common/filesystembase.h                                                                                                                                                                                       
 *      /usr/include/ownCloud/common/ocsynclib.h                                                                                                                                                                                            
 *      /usr/include/ownCloud/libsync/owncloudlib.h                                                                                                                                                                                         
 *      /usr/include/ownCloud/libsync/logger.h                                                                                                                                                                                              
 *      /usr/include/ownCloud/libsync/accessmanager.h                                                                                                                                                                                       
 *      /usr/lib64/cmake/ownCloud/ownCloudConfigVersion.cmake                                                                                                                                                                               
 *      /usr/lib64/cmake/ownCloud/ownCloudConfig.cmake                                                                                                                                                                                      
 *      /usr/lib64/cmake/ownCloud/ownCloudConfig-relwithdebinfo.cmake                                                                                                                                                                       
```

This is mostly caused by https://github.com/owncloud/client/commit/00a2d512b561cc9616e2d01fad01a6f7c565cd1e which changed, for example:
```
INSTALL(
    FILES ${owncloudsync_HEADERS}
    DESTINATION ${INCLUDE_INSTALL_DIR}/${synclib_NAME}
)
```
to
```
INSTALL(
    FILES
        ${CMAKE_CURRENT_BINARY_DIR}/owncloudlib.h
        logger.h
        accessmanager.h
    DESTINATION ${INCLUDE_INSTALL_DIR}/ownCloud/libsync
)
```
dropping the branded name `${synclib_NAME}`. 

### Expected behavior

Branded and non-branded clients can be installed in parallel. 

### Client version number

For example ownCloud-3.2.0.10193 (and beyond).

**Comment by olifre** [comment]
I have developed a patch which fixes the collisions:
[owncloud-client-3.2.0.10193-fix-mirall.patch.txt](https://github.com/owncloud/client/files/11857921/owncloud-client-3.2.0.10193-fix-mirall.patch.txt)

In case this issue is confirmed as a bug (and not intended), I can of course also create a PR to fix it.

**Comment by fmoc** [comment]
Thank you for your detailed bug report. Bug confirmed. The issue persists in `master`, too.

Your patch fixes the problem on the surface, but there are more locations that need to be fixed. I'm working on a PR for `master`, which you can adapt to work on the version you're trying to build.

By the way, 3.2.x is pretty old. We just released 4.1.0. (Please note that this release also needs your patch, though.)
